### DISCUSSION TITLE
Feature Request - Export .intune file for Win32 apps

### DISCUSSION THREAD

**Discussion by dridley23** [Micke-K/IntuneManagement#55] [Ideas]
Would be great if it could do this.

**Comment by Micke-K** [comment]
I couldn't agree more. I ask Microsoft for to add support for this 3 years ago. 

An intunewin is just a zip file but it is encrypted during the upload. The encryption keys are not available via APIs. The script has code to download the intunewin file but without the encryption keys, you can't unpack it. 

Oliver found a way to get the encryption keys but it requires installing the app on a client. See [this blogpost](https://oliverkieselbach.com/2022/03/30/ime-debugging-and-intune-win32-app-decoding-part-2/). 

Rudy Ooms also has a blog post about it [here](https://call4cloud.nl/2022/03/honest-win32app-thief/). 

The best way would be if Microsoft added an API to get the encryption keys if you have ReadWrite permissions to Apps. Or other options like create an additional permissions just for key download, store keys in an Azure Key Vault etc.

**Comment by dridley23** [comment]
Spit balling for a bit @Micke-K ...
...what if the import feature prompted for the ".intune" file so it could auto upload it. OR even better if there was a custom folder we could place the files into prior so your import tool looks for it in there and auto-imports.

Currently we need to click on each app, select view, and then upload the file manually for each app. I'm just trying to think how we could automate it (partially at least). Because, it seems silly to import the app if the intune file isn't available.

**Reply by Micke-K** [reply]
It already exists. See [Discussion 58](https://github.com/Micke-K/IntuneManagement/discussions/58)
